app/src/backend/NotificationService.ts

```diff
@@ -0,0 +1,48 @@
+import { smtpOptions } from "@/lib/email";
+import nodemailer, { Transporter } from "nodemailer";
+
+interface EmailOptions {
+  to: string;
+  subject: string;
+  text: string;
+  html: string;
+}
+
+interface SendEmailResponse {
+  success: boolean;
+  messageId?: string;
+  error?: any;
+}
+
+class NotificationService {
+  private transporter: Transporter;
+  constructor() {
+    this.transporter = nodemailer.createTransport({ ...smtpOptions });
+  }
+
+  async sendEmail({
```

Comment:
Should be static so it can be called like this: `await NotificationService.sendEmail(...);`

Also this should handle template rendering with react-email.